ring/ring_test.go

```diff
@@ -627,27 +628,31 @@ func TestRing_Get_ZoneAwareness(t *testing.T) {
 		"should succeed if there are enough instances per zone on RF = 3": {
 			numInstances:         16,
 			numZones:             3,
+			totalZones:           3,
 			replicationFactor:    3,
 			zoneAwarenessEnabled: true,
 			expectedInstances:    3,
 		},
 		"should fail if there are instances in 1 zone only on RF = 3": {
```

Comment:
This test will fail with the code changes in this PR _unless_ we create at least a single instance in each zone (without tokens) below. The code previously assumed that replication factor was equivalent to the total number of zones. However, that's not the case anymore since replication factor can be higher than the number of zones. Thus, we need to create an instance in each zone so that `ringTokensByZone` gets updated and has the correct number of zones.